cmd/sourced/docker/docker.go

```diff
@@ -0,0 +1,83 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types/filters"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+)
+
+var ErrNotFound = errors.New("container not found")
+
+// GetClient returns a docker client if all checks pass.
+// This function performs three checks:
+//   1. checks that docker is installed and running properly,
+//   2. checks that the user is not running docker toolbox.
+//   3. checks thatf the client api version is supported by the docker engine,
+func GetClient() (*client.Client, error) {
+	// This will fail in case of bad response from the daemon or in
+	// case of docker not installed/running
+	c, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return nil, err
+	}
+
+	var info types.Info
+	// Get information from running daemon to check whether is running
+	// docker toolbox
+	info, err = c.Info(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	if strings.Contains(strings.ToLower(info.OperatingSystem), "boot2docker") {
+		return nil, fmt.Errorf("Docker Toolbox is not supported")
```

Comment:
can you remind me why we don't support Docker Toolbox?
If we don't interact with docker directly but only through `docker-compose` it should work, no?

Reply:
mhmm... Actually, I copy-pasted the code from engine as you without thinking at the reason. But that's definitely something to rethink about. I don't remember very well the reason, AFAIR it has something to do with Windows support, but I have to recheck it.
If this limitation doesn't apply by using only `docker-compose`, then for sure it's better to avoid using the client and parse the stdout.

Reply:
@carlosms @dpordomingo do you perhaps the remember the reason? I don't remember whether we _really_ do not support docker toolbox or whether we just decided to not support it.